api/config/v1alpha1/envoygateway_types.go

```diff
@@ -107,6 +108,11 @@ type RateLimit struct {
 	// database backend used by the rate limit service to store
 	// state associated with global ratelimiting.
 	Backend RateLimitDatabaseBackend `json:"backend"`
+
+	// DeploymentResources required by the ratelimit deployment container.
```

Comment:
I did a version like this 

```golang
// RateLimit defines the configuration associated with the Rate Limit Service
// used for Global Rate Limiting.
type RateLimit struct {
	// Backend holds the configuration associated with the
	// database backend used by the rate limit service to store
	// state associated with global ratelimiting.
	Backend RateLimitDatabaseBackend `json:"backend"`
	// Deployment defines the desired state of the Envoy ratelimit deployment resource.
	// If unspecified, default settings for the manged Envoy ratelimit deployment resource
	// are applied.
	//
	// +optional
	Deployment *KubernetesDeploymentSpec `json:"deployment,omitempty"`
}
```

but I finally changed my mind, I think the replicas setting for the ratelimit is unnecessary, just set the `REDIS_TYPE` to `cluster`, redis to handle more traffic.

I'll wait for the conclusion of the community meeting and refactor this.

Reply:
chatted with @haq204 @AliceProxy in the community meeting today and we prefer
```
 provider:
    kubernetes:
      rateLimitDeployment:
        replicas: 2
```
to reduce the number of nested structures

but prefer adding this if actually needed by real users today

Reply:
I will go on after #1232 merged

Reply:
> @qicz are you using ratelimiting ? if so are you hitting scale issues ?

It is not used now, but it will be used in the future. and hope it can be configured like envoy deployment, auth can also be configured